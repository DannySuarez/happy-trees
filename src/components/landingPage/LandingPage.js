import React from 'react';
import PropTypes from 'prop-types';
import styles from './landingPage.css';

function LandingPage({ handleSubmit, nickname, handleUpdate }) {
  return (
    <div className={styles.Full}>

      <h1 className={styles.Logo}>Happy Trees</h1>

      <div className={styles.LandingPage}>
        <h1>Choose Nickname:</h1>

        <div className={styles.Form}>
          <form onSubmit={handleSubmit}>
            <input 
              type="text" 
              name="nickname" 
              value={nickname} 
              placeholder='nickname...' 
              onChange={handleUpdate}
            /> 
            <br/>
            <button>Enter</button>
          </form>

        </div>

      </div>
      
    </div>
  );
}

LandingPage.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  nickname: PropTypes.string.isRequired,
  handleUpdate: PropTypes.func.isRequired
};

export default LandingPage;
