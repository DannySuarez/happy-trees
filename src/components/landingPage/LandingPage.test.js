import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage';

describe('LandingPage component', () => {
  it('renders App', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
