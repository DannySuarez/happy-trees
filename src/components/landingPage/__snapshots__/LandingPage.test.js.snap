// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders App 1`] = `
<div
  className="LandingPage"
>
  <h1>
    Choose Nickname:
  </h1>
  <div
    className="Form"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="nickname"
        onChange={[Function]}
        placeholder="nickname..."
        type="text"
        value=""
      />
      <button>
        Enter
      </button>
    </form>
  </div>
</div>
`;
